validation-tool-backend/src/users/users.service.ts

```diff
@@ -74,7 +74,18 @@ export class UsersService {
         AND password = HASHBYTES('SHA2_256', @Password)`);
     if (result.recordset && result.recordset.length > 0) {
       user = result.recordset[0] as User;
+      const roles = await pool.request()
+        .input('SUSD_USER_ID', Int, user.id)
+        .query('SELECT run_id, roles FROM reviewer where susd_user_id = @SUSD_USER_ID');
+      if (roles.recordset && roles.recordset.length > 0) {
+        const privileges = roles.recordset;
+        privileges.forEach(element => {
```

Comment:
I don't know if I understand ts well enough, but it seems that you are not storing the run_id with the privileges. Not other information about the aganecy_run, which I think will be useful for the UI at some point. For example for users who can review multiple runs, one could show snippets in different tabs or who-knows-how one might wish distinguish snippets from different runs. same for admins who can monitor/configure the validation progress of multiple runs.

Reply:
Hey @glemson , the access_token has the run_id, this is what the login endpoint will return:

`{
  "id": "1",
  "email": "mtaghiza@jhu.edu",
  "first_name": "Manuchehr",
  "last_name": "Taghizadeh-Popp",
  "privileges": [
    {
      "run_id": "1",
      "roles": [
        "REVIEWER"
      ]
    },
    {
      "run_id": "3",
      "roles": [
        "REVIEWER"
      ]
    }
  ],
  "iat": 1666486082,
  "exp": 1666529282
}`

Reply:
ah ok, thanks. indeed this was my lack of knowledge.
does the UI already know how to link the run_id to human readable info about the agency runs?